addons/crm/models/crm_lead.py

```diff
@@ -751,7 +752,9 @@ def action_set_won(self):
         # group the leads by team_id, in order to write once by values couple (each write leads to frequency increment)
         leads_by_won_stage = {}
         for lead in self:
-            stage_id = lead._stage_find(domain=[('is_won', '=', True)])
+            stage_id = lead._stage_find(domain=['&', ('is_won', '=', True), ('sequence', '>', lead.stage_id.sequence)])
+            if not stage_id:
+                stage_id = lead._stage_find(domain=['&', ('is_won', '=', True), ('sequence', '<=', lead.stage_id.sequence)], order="sequence desc")
```

Comment:
Why not searching directly by sequence desc ? It would allow to search only once. Indeed if greater sequence does not return anything you search for lower sequence. Seems like we could simply order directly by sequence desc.

Reply:
As the issue occured when there was multiple "Won stages" (one before current stage, and one after), I assumed there could be multiple "Won stages" after the current stage. 
When the current stage is set as won, by ordering with a sequence asc (default), I can ensure that the selected stage is the one closest to the current stage. If this idea is wrong/not useful I will do as you say.

Reply:
Is it required to find "first one after or last one before" ?

Reply:
endless debate about what should we do with won stage (all at the end, or allow to have 'intermediate' won stage in fonction of the team for example..)
For me, all won stage should be at the end but maybe @LNAodoo does not agree ?

Reply:
I agree with @tde-banana-odoo. As usual, he is right ! 
To ensure PLS is not fucked up, a lead worked as won should always be sent to the last won stage. (so stat of the previous stages are correctly computed).
If there is no won stage after the current opportunity's stage, it will be sent to the closest won stage before it.
PLS will be fucked up anyway, but that will be the fault of the fucked up user that configured the stages like a 'manche'.
Maybe later we could add some constraints to ensure won stages are always at the end of the pipe.But that's another problem.
So TDE's remark is valid and I "plussoie" it.